user-guide/Advanced_Functionality/Swarming/PrepareSwarming.md

```diff
@@ -0,0 +1,201 @@
+---
+uid: SwarmingPrepare
+---
+
+# Preparing your system for Swarming (Alarm ID ranges)
+
+When the Swarming feature is enabled, alarm references need to be globally unique within the cluster. For that reason, alarm IDs will be generated in a different way than in a system without Swarming. It is important to make sure that scripts and connectors are compatible with this to be able to successfully work with a Swarming-enabled setup.
+
+> [!NOTE]
+> The changes mentioned on this page require **DataMiner 10.5.1/10.5.0 or higher**. Note that while legacy alarm references may still be used in DataMiner 10.5.1 or higher, they must not be used on systems where the Swarming feature is enabled.
+
+## Terminology
+
+- **Alarm Event**: An object indicating that either a new alarm came up, an existing event was updated (severity changed, properties updated, acknowledged, ...) or was cleared.
+- **Root Event**: Initial alarm event, as one is created when there was previously no alarm but now there is.
+- **Alarm Tree**: A logical alarm, combining a history of alarm events for the same alarm. A tree will always have a root alarm event plus zero or more alarm event updates that have happened over time. Each update links back to the original root alarm event as well as to the previous alarm event and has a unique alarm ID number (e.g. `444`) within that tree.
+- **Alarm Tree ID**: Reference which can be used to uniquely refer to an alarm tree without having to refer to individual alarm events. For example, when adding a comment to an alarm tree, one can send the alarm tree ID to the server to apply the update to the most recent state of that alarm tree. An ID looks like `111/222/333` where `111/222` is a reference to the element involved. The Alarm Tree ID is the preferred way to reference alarm trees.
+- **Root Alarm ID**: Legacy number (e.g. `333`) indicating a specific alarm tree. Not guaranteed to be unique across elements or agents.
+- **Alarm ID**: Number referencing an alarm event within an alarm tree (e.g. `444`). Not to be used on its own. Can be used in combination with an alarm tree ID in cases where an individual alarm event needs to be referenced. Try to avoid these ([more info](#alarmid-references)). For root alarm events, the alarm ID is identical to the root alarm ID.
+- **Hosting Agent ID**: DataMiner ID of the agent on which the element owning the alarm event/tree is currently hosted.
```

Comment:
not completely/always true right?

Its the agent that was hosting the element when the alarm was created?

Reply:
updated